neo.UnitTests/UT_MemoryPool.cs

```diff
@@ -0,0 +1,214 @@
+
+                var mockStore = new Mock<Store>();
+
+                var defaultTx = CreateRandomHashInvocationTransaction();
+                defaultTx.Outputs = new TransactionOutput[1];
+                defaultTx.Outputs[0] = new TransactionOutput
+                {
+                    AssetId = Blockchain.UtilityToken.Hash,
+                    Value = new Fixed8(1000000), // 0.001 GAS (enough to be a high priority TX
+                    ScriptHash = UInt160.Zero // doesn't matter for our purposes.
+                };
+
+                mockStore.Setup(p => p.GetBlocks()).Returns(new TestDataCache<UInt256, BlockState>());
+                mockStore.Setup(p => p.GetTransactions()).Returns(new TestDataCache<UInt256, TransactionState>(
+                    new TransactionState
+                    {
+                        BlockIndex = 1,
+                        Transaction = defaultTx
+                    }));
+
+                mockStore.Setup(p => p.GetAccounts()).Returns(new TestDataCache<UInt160, AccountState>());
+                mockStore.Setup(p => p.GetUnspentCoins()).Returns(new TestDataCache<UInt256, UnspentCoinState>());
+                mockStore.Setup(p => p.GetSpentCoins()).Returns(new TestDataCache<UInt256, SpentCoinState>());
+                mockStore.Setup(p => p.GetValidators()).Returns(new TestDataCache<ECPoint, ValidatorState>());
+                mockStore.Setup(p => p.GetAssets()).Returns(new TestDataCache<UInt256, AssetState>());
+                mockStore.Setup(p => p.GetContracts()).Returns(new TestDataCache<UInt160, ContractState>());
+                mockStore.Setup(p => p.GetStorages()).Returns(new TestDataCache<StorageKey, StorageItem>());
+                mockStore.Setup(p => p.GetHeaderHashList()).Returns(new TestDataCache<UInt32Wrapper, HeaderHashList>());
+                mockStore.Setup(p => p.GetValidatorsCount()).Returns(new TestMetaDataCache<ValidatorsCountState>());
+                mockStore.Setup(p => p.GetBlockHashIndex()).Returns(new TestMetaDataCache<HashIndexState>());
+                mockStore.Setup(p => p.GetHeaderHashIndex()).Returns(new TestMetaDataCache<HashIndexState>());
+                mockStore.Setup(p => p.GetSnapshot()).Returns(mockSnapshot.Object);
+
+                Console.WriteLine("initialize NeoSystem");
+                TheNeoSystem = new NeoSystem(mockStore.Object); // new Mock<NeoSystem>(mockStore.Object);
+            }
+
+            // Create a MemoryPool with capacity of 100
+            _unit = new MemoryPool(TheNeoSystem, 100);
+
+            // Verify capacity equals the amount specified
+            _unit.Capacity.ShouldBeEquivalentTo(100);
+
+            _unit.VerifiedCount.ShouldBeEquivalentTo(0);
+            _unit.UnVerifiedCount.ShouldBeEquivalentTo(0);
+            _unit.Count.ShouldBeEquivalentTo(0);
+        }
+
+        private Transaction CreateRandomHashInvocationTransaction()
```

Comment:
The random value shouldn't matter unless it managed to generate a duplicate, but that is insanely unlikely for a 16 byte random which is how I'm using it. I'll still likely switch to a seed so it will always be deterministic for other future users of random.